doc/Command-Reference.md

```diff
@@ -7961,8 +7962,292 @@ Go Back To [Beginning of the document](#) or [Beginning of this section](#waterm
+  --------------  ---------------------------  ----------------------------  ---------  ---------
+  database        docker-database              SONiC database package        1.0.0      Built-In
+  dhcp-relay      docker-dhcp-relay            SONiC dhcp-relay package      1.0.0      Installed
+  fpm-frr         docker-fpm-frr               SONiC fpm-frr package         1.0.0      Built-In
+  lldp            docker-lldp                  SONiC lldp package            1.0.0      Built-In
+  macsec          docker-macsec                SONiC macsec package          1.0.0      Built-In
+  mgmt-framework  docker-sonic-mgmt-framework  SONiC mgmt-framework package  1.0.0      Built-In
+  nat             docker-nat                   SONiC nat package             1.0.0      Built-In
+  pmon            docker-platform-monitor      SONiC pmon package            1.0.0      Built-In
+  radv            docker-router-advertiser     SONiC radv package            1.0.0      Built-In
+  sflow           docker-sflow                 SONiC sflow package           1.0.0      Built-In
+  snmp            docker-snmp                  SONiC snmp package            1.0.0      Built-In
+  swss            docker-orchagent             SONiC swss package            1.0.0      Built-In
+  syncd           docker-syncd-mlnx            SONiC syncd package           1.0.0      Built-In
+  teamd           docker-teamd                 SONiC teamd package           1.0.0      Built-In
+  telemetry       docker-sonic-telemetry       SONiC telemetry package       1.0.0      Built-In
+  ```
+
+**sonic-package-manager repository add**
+
+This command will add a new entry in the package database. The package has to be *Not Installed* in order to be removed from package database. *NOTE*: this command requires elevated (root) privileges to run.
+
+- Usage:
+  ```
+  Usage: sonic-package-manager repository add [OPTIONS] NAME REPOSITORY
+
+    Add a new repository to database.
+
+  Options:
+    --default-reference TEXT  Default installation reference.
+    --description TEXT        Optional package entry description.
+    --help                    Show this message and exit.
+  ```
+- Example:
+  ```
+  admin@sonic:~$ sudo sonic-package-manager repository add \
+    cpu-report azure/sonic-cpu-report --default-reference 1.0.0
+  ```
+
+**sonic-package-manager repository remove**
+
+This command will remove an entry from the package database. *NOTE*: this command requires elevated (root) privileges to run.
+
+- Usage:
+  ```
+  Usage: sonic-package-manager repository remove [OPTIONS] NAME
+
+    Remove package from database.
+
+  Options:
+    --help  Show this message and exit.
+  ```
+- Example:
+  ```
+  admin@sonic:~$ sudo sonic-package-manager repository remove cpu-report
+  ```
+
+**sonic-package-manager install**
+
+This command pulls and installs package on SONiC host. *NOTE*: this command requires elevated (root) privileges to run.
+
```

Comment:
Make it explicit "this is only for default/added packages"

would you require "-f", when  you are downgrading ?

What would be the user faced behavior, if the package is currently running ?

Is there any option for user to revert back to what he had before ?

Reply:
"this is only for default/added packages" - what does it mean? You mean that the user first needs to add a repository?

"would you require "-f", when you are downgrading ?" - yes, please see the comment block in the code with some thoughts on  downgrade - https://github.com/Azure/sonic-utilities/pull/1527/files#diff-a8e54e07d62d4629cbfd9ca9c289208aa437f0d8f722a8f8b363df9c38b315e6R475-R488

"What would be the user faced behavior, if the package is currently running ?". If a package is currently running, meaning it is installed, installing the same package again (or different version of the same package) will be blocked https://github.com/Azure/sonic-utilities/pull/1527/files#diff-a8e54e07d62d4629cbfd9ca9c289208aa437f0d8f722a8f8b363df9c38b315e6R322-R325


"Is there any option for user to revert back to what he had before ?" - no, user is required to downgrade with force.

Reply:
BTW, the default versions that came with the image are recorded in "default_reference" in packages.json, so user can always look for default and reinstall the one that was planned to be used for his sonic image

Reply:
We can add a command to reset the package to a default version:
```
reset-default x == upgrade x==<default_reference>
```